src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -96,98 +104,194 @@ export default class Preview extends Setup {
-        return new Promise<void>((resolve, reject) => {
-            const isValid: boolean = compName
-                ? compName.trim().length > 0
-                : false;
-            if (isValid === false) {
-                this.logger.debug('Invoked validateComponentName in preview');
-                reject(
+    public async validateAdditionalInputs(): Promise<void> {
+        const platform = this.flags.platform;
+
+        const compName = CommandLineUtils.resolveFlag(
+            this.flags.componentname,
+            ''
+        ).trim();
+
+        const targetApp = CommandLineUtils.resolveFlag(
+            this.flags.targetapp,
+            PreviewUtils.BROWSER_TARGET_APP
+        );
+
+        const configFileName = CommandLineUtils.resolveFlag(
+            this.flags.configfile,
+            ''
+        ).trim();
+
+        const hasConfigFile =
+            configFileName.length > 0 && fs.existsSync(configFileName);
+
+        const isBrowserTargetApp = PreviewUtils.isTargetingBrowser(targetApp);
+
+        const isValidCompName = compName.length > 0;
+
+        this.logger.debug('Validating Preview command inputs.');
+
+        // check if user provided a config file when targetapp=browser
+        // and warn them that the config file will be ignored.
+        if (isBrowserTargetApp && hasConfigFile) {
+            this.logger.warn(
+                messages.getMessage('ignoringConfigFileFlagDescription')
+            );
+        }
+
+        if (isValidCompName === false) {
+            return Promise.reject(
+                new SfdxError(
+                    messages.getMessage(
+                        'error:invalidComponentNameFlagsDescription'
+                    ),
+                    'lwc-dev-mobile',
+                    Preview.examples
+                )
+            );
+        }
+
+        if (isBrowserTargetApp === false && hasConfigFile === false) {
+            return Promise.reject(
+                new SfdxError(
+                    messages.getMessage(
+                        'error:invalidConfigFile:missingDescription',
+                        [configFileName]
+                    ),
+                    'lwc-dev-mobile',
+                    Preview.examples
+                )
+            );
+        }
+
+        if (isBrowserTargetApp === false && hasConfigFile === true) {
+            // 1. validate config file against schema
+            const validationResult = await PreviewUtils.validateConfigFileWithSchema(
+                configFileName,
+                configSchema
+            );
+            if (validationResult.passed === false) {
+                return Promise.reject(
                     new SfdxError(
                         messages.getMessage(
-                            'error:invalidInputFlagsDescription'
+                            'error:invalidConfigFile:genericDescription',
+                            [configFileName, validationResult.errorMessage]
```

Comment:
At least for error messages related to configuration, let's do a `path.resolve(configFileName)` for the message. In the error case, I don't want one of the ambiguities to be uncertainty of the path to the file.